cdc/api/v2/api.go

```diff
@@ -14,25 +14,71 @@
 package v2
 
 import (
+	"net/http"
+
 	"github.com/gin-gonic/gin"
 	"github.com/pingcap/tiflow/cdc/api/middleware"
 	"github.com/pingcap/tiflow/cdc/capture"
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/cdc/owner"
+	"github.com/pingcap/tiflow/pkg/upstream"
 )
 
 // OpenAPIV2 provides CDC v2 APIs
 type OpenAPIV2 struct {
 	capture *capture.Capture
+	// use for unit test only
+	testStatusProvider owner.StatusProvider
```

Comment:
I think it's not the best way to mock the behavior of `OpenAPIV2`.

We can use failpoint into the method `statusProvider`, eg:
```go
func (h *OpenAPIV2) statusProvider() owner.StatusProvider {
	failpoint.Inject("mockStatusProvider", func(){
		failpoint.Return(newMockStatusProvider())
	})
	return h.capture.StatusProvider()
}
```

To avoid pollute the definition of type `OpenAPIV2`.

Reply:
Thx & Fixed. For now, I erased the dependence on StatusProvider by removal since tso_test didn't use it.